### ISSUE TITLE
Better errors for `process`

### ISSUE THREAD

**Issue by gaborcsardi** [r-lib/processx#197]
This is trickier, because by default we do not even record the stderr. But we should at least have a reasonable way of saying that we want some error messages.

**Comment by gaborcsardi** [comment]
There is probably no good canned solution here right now. We could have a `communicate()`-like method, like Python, and then that could have a canned stderr "handler" that collects the last N lines or bytes of stderr.

**Comment by skiaphrene** [comment]
Dear processx team,

After developing and installing my own private R package on my Mac computer, I am currently trying to install it on a Windows computer. For some reason I am getting an error message that I tracked down to processx, and that seems like may fit in this thread:

> Error: Failed to install 'lipidR' from GitHub:
> Command not found @win/processx.c:977

I have no idea why processx is being invoked during my installation (possibly through Rtools?), but that line in the processx code would be called when an application is not found. 

SUGGESTION: Would it be possible to **add more verbose output to the R_THROW_ERROR call on processx.c line 977**, such as adding the name of the application it searched for, the path it explored, etc? This would help debugging cases like this, and from googling the error, I noticed that there was a fair amount of users confused with similar laconic error messages (possibly from earlier versions as the line numbers don't match up), so as far as I can tell it's not just me.

Thanks in advance and best regards,

-- Alex
